internal/supervisor/server/server.go

```diff
@@ -465,11 +544,73 @@ func runSupervisor(ctx context.Context, podInfo *downward.PodInfo, cfg *supervis
 	plog.Debug("supervisor started")
 	defer plog.Debug("supervisor exiting")
 
+	// Run the server. Its post-start hook will start the controllers.
```

Comment:
Lets manually make sure that shutdown still occurs cleanly (I believe it should).